custom-targets/vertex-ai/model-deployer/vertexai.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"google.golang.org/api/aiplatform/v1"
+	"google.golang.org/api/option"
+	"os"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+// deployModelFromManifest loads the file provided in `path` and returns the parsed DeployModelRequest
+// from the data.
+func deployModelFromManifest(path string) (*aiplatform.GoogleCloudAiplatformV1DeployModelRequest, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("error reading manifest file: %v", err)
+	}
+
+	deployModelRequest := &aiplatform.GoogleCloudAiplatformV1DeployModelRequest{}
+
+	if err = yaml.Unmarshal(data, deployModelRequest); err != nil {
+		return nil, fmt.Errorf("unable to parse deploy model deployModelRequest from manifest file: %v", err)
+	}
+
+	return deployModelRequest, nil
+}
+
+// determinePreviousModel queries the provided Vertex AI endpoint to determine the model that was previously
+// deployed.
+func determinePreviousModel(service *aiplatform.Service, endpointName, currentModel string) (string, error) {
```

Comment:
nit: PreviousDeployedModel right?

Reply:
No the previous deployed model is obtained by querying the endpoint. `currentModel` is the model that is being canaried.